### PR TITLE
Add automatic weeding to weeds ability

### PR DESCRIPTION
**Pull request by Walarks**
## About The Pull Request
GIves the "plant weeds" ability the option to turn on automatic weeding. This makes the xeno try to use the ability when they move more than 3 tiles away from the last planted node.

## Why It's Good For The Game
Saves worker xeno sanity from repetitive manual labor.

## Changelog
:cl:
qol: add to the weeds menu an option to automatically use the ability when moving
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Neat

**Comment by ilikesaying** [comment]
now add a weed-whacker

**Comment by Derrick2Electricboogaloo** [comment]
Okay now where is engineer auto cade placement ability. God forbid we have to put in the manual work this is an unrealistic expectation.

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 1]
```diff
@@ -77,20 +82,47 @@
...
+	if(auto_weeding)
+		UnregisterSignal(owner, COMSIG_MOVABLE_MOVED)
+		UnregisterSignal(owner, COMSIG_MOB_DEATH)
+		auto_weeding = FALSE
+		to_chat(owner, "<span class='notice'>We will no longer automatically plant weeds.</span>")
+		return
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, .proc/weed_on_move)
+	RegisterSignal(owner, COMSIG_MOB_DEATH, .proc/toggle_auto_weeding)
+	auto_weeding = TRUE
+	to_chat(owner, "<span class='notice'>We will now automatically plant weeds.</span>")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:112]: use span_class() macros

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 2]
```diff
@@ -77,20 +82,47 @@
...
-			break
-	to_chat(owner, "<span class='notice'>We will now spawn <b>[weed_choice]\s</b> when using the plant weeds ability.</span>")
+	if(weed_choice == AUTOMATIC_WEEDING)
+		toggle_auto_weeding()
+	else
+		for(var/obj/effect/alien/weeds/node/weed_type_possible AS in GLOB.weed_type_list)
+			if(initial(weed_type_possible.name) == weed_choice)
+				weed_type = weed_type_possible
+				break
+		to_chat(owner, "<span class='notice'>We will now spawn <b>[weed_choice]\s</b> when using the plant weeds ability.</span>")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:97]: span macros

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 3]
```diff
@@ -77,20 +82,47 @@
...
 
+///Toggles automatic weeding
+/datum/action/xeno_action/activable/plant_weeds/proc/toggle_auto_weeding()
+	SIGNAL_HANDLER
+	if(auto_weeding)
+		UnregisterSignal(owner, COMSIG_MOVABLE_MOVED)
+		UnregisterSignal(owner, COMSIG_MOB_DEATH)
+		auto_weeding = FALSE
+		to_chat(owner, "<span class='notice'>We will no longer automatically plant weeds.</span>")
+		return
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:108]: span

**Review by TiviPlus — CHANGES_REQUESTED** [review]
+ discord comment
after you fix conflict should be good for tmc

File: code/modules/mob/living/carbon/xenomorph/abilities.dm (hunk 2 above)
Comment (Walarks) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: added span macros

File: code/modules/mob/living/carbon/xenomorph/abilities.dm (hunk 3 above)
Comment (Walarks) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: ditto

File: code/modules/mob/living/carbon/xenomorph/abilities.dm (hunk 1 above)
Comment (Walarks) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm]: added span macros

**Pull request merged**